app/src/main/java/org/jellyfin/androidtv/details/DetailsActivity.kt

```diff
@@ -0,0 +1,56 @@
+package org.jellyfin.androidtv.details
+
+import android.os.Bundle
+import android.util.Log
+import androidx.fragment.app.FragmentActivity
+import androidx.leanback.app.BackgroundManager
+import com.squareup.picasso.Picasso
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.TvApp
+import org.jellyfin.androidtv.util.apiclient.asEpisode
+import org.jellyfin.androidtv.util.apiclient.getItem
+import org.jellyfin.apiclient.model.dto.BaseItemDto
+import org.jellyfin.apiclient.model.dto.BaseItemType
+import org.jellyfin.apiclient.model.dto.ImageOptions
+
+class DetailsActivity : FragmentActivity() {
+	private lateinit var fragment: BaseDetailsFragment
+
+	override fun onCreate(savedInstanceState: Bundle?) {
+		super.onCreate(savedInstanceState)
+
+		val id = intent.getStringExtra("id")
+		if (id == null) finish()
+
+		BackgroundManager.getInstance(this).attach(window)
+
+		GlobalScope.launch(Dispatchers.Main) {
+			val baseItem = getBaseItemDtoForID(id) ?: return@launch
+
+			when (baseItem.baseItemType!!) {
```

Comment:
You can actually do something like:
```kotlin
fragment = when(){....}
```

As long as each case returns something it should work